### PR TITLE
Filter institute cases using multiple HPO terms - Fixed gene variants search with HPO terms

### LINKED ISSUE #2690: Filter for multiple phenotypes
Latest from Scout user meeting. Allow multiple phenotype search terms for cases via case query generation, and then unless there are unforeseen complications by extension also for gene-variants as well as dashboard where the same query generation is used.

### LINKED ISSUE #3073: Gene variants search is not working when filtering by HPO terms
Local example:
Demo case has default HPO terms: `HP:0001298` and `HP:0001250`.

When filtering gene variants using gene POT1, rank score >5 for demo case, some results are returned:

image

When adding an HPO term in the filtering (for instance HP:0001298), no variants are returned:

image

### PR DESCRIPTION
**Pull request by northwestwitch**
- Allow filtering institute cases using multiple HPO terms: fix #2690
- Fix searching gene variants specifying one or more HPO terms: fix #3073

**Preparing for test**:
1. Locally, load a second case in scout demo: `scout --demo load case scout/demo/643595.config.yaml`
2. Assign one or more HPO terms to this case, for instance: HP:0012444
3. Note that demo case 643594 has already assigned a couple of phenotypes

**Testing**:
1. On main branch, make sure that filtering cases by multiple HPO terms doesn't work in:
- Cases page (http://localhost:5000/cust000/cases)
- Dashboard (http://localhost:5000/dashboard)
- Gene variants (http://localhost:5000/cust000/gene_variants). In this page even searching with one term is broken

2. Switch to this branch and try the pages above by filtering with one phenotype from case 643594 and one phenotype from case 643595. All the pages should return meaningful results (from 2 cases)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN, CR

### CODE REVIEW

File: scout/adapter/mongo/query.py [hunk 1]
```diff
@@ -129,9 +129,9 @@ def build_variant_query(
 
         if mongo_case_query != {}:
             mongo_case_query["owner"] = institute_id
-            LOG.debug("Search cases for selection set, using query {0}".format(select_case_obj))
-            select_case_obj = self.case_collection.find(mongo_case_query)
-            select_cases = [case_id.get("display_name") for case_id in select_case_obj]
+            LOG.debug("Search cases for selection set, using query {0}".format(mongo_case_query))
+            select_case_objs = self.case_collection.find(mongo_case_query)
+            select_cases = [case_id.get("_id") for case_id in select_case_objs]
```
Comment (northwestwitch) [inline scout/adapter/mongo/query.py:134]: This is the fix for #3073

File: scout/adapter/mongo/case.py [hunk 2]
```diff
@@ -168,12 +168,16 @@ def _populate_name_query(self, query, name_query, owner=None, collaborator=None)
 
         if query_field == "exact_pheno":
             if query_term != "":
-                query["phenotype_terms.phenotype_id"] = query_term
+                # User might have provided multiple query terms
+                query["phenotype_terms.phenotype_id"] = {
+                    "$in": [term for term in query_term.replace(" ", "").split(",")]
```
Comment (northwestwitch) [inline scout/adapter/mongo/case.py:173]: This is the fix for #2690

File: scout/constants/case_tags.py [hunk 3]
```diff
@@ -134,7 +134,7 @@
 CASE_SEARCH_TERMS = {
     "case": {"label": "Case or Individual Name", "prefix": "case:"},
     "exact_pheno": {
-        "label": "HPO Term",
+        "label": "HPO Terms",
```
Comment (dnil) [inline scout/constants/case_tags.py:137]: ✅

File: scout/adapter/mongo/query.py (hunk 1 above)
Comment (dnil) [inline scout/adapter/mongo/query.py:134]: Nice! And that was there all this time... 🙄

File: scout/server/blueprints/institutes/static/form_scripts.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ var sel = document.getElementById('search_type');
 
 var selectHelper = {
   "case:": "example:18201",
-  "exact_pheno:": "example:HP:0001166",
+  "exact_pheno:": "example:HP:0001166,HP:0001250,..",
```
Comment (dnil) [inline scout/server/blueprints/institutes/static/form_scripts.js:6]: 👍

Comment (dnil) [reply scout/server/blueprints/institutes/static/form_scripts.js:6]: Full ellipsis!
```suggestion
  "exact_pheno:": "example:HP:0001166,HP:0001250,...",
```

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
We really could use some cutoffs for HPO-similarity as well, but that is another project.

**Pull request merged**